File: src/main/java/net/hectus/neobb/turn/person_game/categorization/BuffCategory.java
package net.hectus.neobb.turn.person_game.categorization;

import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.format.TextColor;
import org.bukkit.Material;

public interface BuffCategory extends Category, BuffFunction {
    @Override
    default TextColor categoryColor() {
        return NamedTextColor.LIGHT_PURPLE;
    }

    @Override
    default int categoryMaxPerDeck() {
        return 9;
    }

    @Override
    default Material categoryItem() {
        return Material.CAKE;
    }

    @Override
    default String categoryName() {
        return "buff";
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/categorization/BuffCategory.kt
package net.hectus.neobb.modes.turn.person_game.categorization

import net.hectus.neobb.modes.turn.default_game.attribute.function.BuffFunction
import net.kyori.adventure.text.format.NamedTextColor
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Material

interface BuffCategory : Category, BuffFunction {
    override val categoryColor: TextColor get() = NamedTextColor.LIGHT_PURPLE
    override val categoryMaxPerDeck: Int get() = 9

    override val categoryItem: Material get() = Material.CAKE
    override val categoryName: String get() = "buff"
}
